client/app/components/group/_pickupList/pickupList.controller.js

```diff
@@ -191,6 +191,23 @@ class PickupListController {
       }
     });
   }
+
+  joinPickup(pickupId) {
```

Comment:
Can you move this logic completely into pickupListItem? The `collector_ids` can also be changed from there.
Also, have a look at the JS Array.find() function, it can simplify this.